### PR TITLE
feat: port upstream `vite-node` dev server implementation

### PR DESCRIPTION
**Pull request by wattanx**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
I will port vite-node-server from Nuxt 3.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by antfu** [comment]
In general it looks good to me. Can you point me out the changes you made to the original code when doing that port, if any? It would be helpful to review. Thanks

File: packages/bridge/src/vite/vite-node.ts [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+  const manifest = normalizeViteManifest({
+    '@vite/client': {
+      file: '@vite/client',
+      css,
+      module: true,
+      isEntry: true
+    },
+    'entry.mjs': {
+      file: 'entry.mjs',
```
Comment (wattanx) [inline packages/bridge/src/vite/vite-node.ts:113]: Changes made to original code.


original code is
```ts
[ctx.entry]: {
  file: ctx.entry
}
```

File: packages/bridge/src/vite/dev-bundler.ts [hunk 2]
```diff
@@ -221,3 +228,36 @@ async function __instantiateModule__(url, urlStack) {
...
+  const viteServer = ctx.ssrServer!
+  const options: TransformOptions = {
+    viteServer,
+    isExternal: createIsExternal(viteServer, ctx.nuxt.options.rootDir)
+  }
+
+  // Build and watch
+  const _doBuild = async () => {
+    const start = Date.now()
+    const { code, ids } = await bundleRequest(options, '/.nuxt/entry.mjs')
```
Comment (wattanx) [inline packages/bridge/src/vite/dev-bundler.ts]: Changes made to original code.

original code is
```ts
const { code, ids } = await bundleRequest(options, ctx.entry)
```

File: packages/bridge/src/vite/manifest.ts [hunk 3]
```diff
@@ -147,3 +147,55 @@ export async function writeClientManifest (clientManifest: any, buildDir: string
...
+    '@vite/client': {
+      isEntry: true,
+      file: '@vite/client',
+      css,
+      module: true,
+      resourceType: 'script'
+    },
+    'entry.mjs': {
+      isEntry: true,
+      file: 'entry.mjs',
```
Comment (wattanx) [inline packages/bridge/src/vite/manifest.ts:166]: Changes made to original code.

original code is
```ts
[ctx.entry]: {
  isEntry: true, // no change
  file: ctx.entry
}
```

File: packages/bridge/src/vite/dev-bundler.ts [hunk 4]
```diff
@@ -221,3 +228,36 @@ async function __instantiateModule__(url, urlStack) {
...
+  const viteServer = ctx.ssrServer!
+  const options: TransformOptions = {
+    viteServer,
+    isExternal: createIsExternal(viteServer, ctx.nuxt.options.rootDir)
+  }
+
+  // Build and watch
+  const _doBuild = async () => {
+    const start = Date.now()
+    const { code, ids } = await bundleRequest(options, '/.nuxt/server.js')
```
Comment (wattanx) [inline packages/bridge/src/vite/dev-bundler.ts:242]: Changes made to original code.

original code is
```ts
const { code, ids } = await bundleRequest(options, ctx.entry)
```

**Review by wattanx — COMMENTED** [review]
I commented on the changes made from the original code!

**Review by danielroe — APPROVED** [review]

**Pull request merged**